crates/router/src/core/payments.rs

```diff
@@ -3569,6 +3929,192 @@ where
+
+    // Since this flow will only be used in the MIT flow, recurring details are mandatory.
+    let recurring_payment_details = payment_data
+        .get_recurring_details()
+        .ok_or(errors::ApiErrorResponse::IncorrectPaymentMethodConfiguration)
+        .attach_printable("Failed to fetch recurring details for mit")?;
+
+    let (mandate_reference_id, card_details) =
+        recurring_payment_details.get_nti_and_card_details_for_mit_flow()?;
+
+    // Currently, only the StraightThrough routing algorithm is supported for the
+    // NTI and card details-based MIT flow.
+    let straight_through_routing_algorithm = match connector_choice {
+        api::ConnectorChoice::StraightThrough(straight_through) => Ok(straight_through),
+        api::ConnectorChoice::SessionMultiple(_) | api::ConnectorChoice::Decide => {
+            Err(errors::ApiErrorResponse::InternalServerError).attach_printable(
+                "Invalid routing rule configured for nti and card details based mit flow",
+            )
+        }
+    }?
+    .parse_value::<api::routing::StraightThroughAlgorithm>("RoutingAlgorithm")
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable("Failed to parse the straight through routing algorithm")?;
+
+    let (routable_connector_choice, should_perform_eligibility_check) =
+        routing::perform_routing_for_single_straight_through_algorithm(
+            &straight_through_routing_algorithm,
+            payment_data.get_creds_identifier(),
+        )
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed execution of straight through routing")?;
+
+    // Filter the network transaction id supported connectors from the routable connector choices.
+    let mit_connector = filter_eligible_connectors(
+        state,
+        payment_data,
+        key_store,
+        routable_connector_choice,
+        should_perform_eligibility_check,
+        eligible_routable_connectors.as_ref(),
+        business_profile,
+    )
+    .await?;
+
+    let mit_first_connector = mit_connector
+        .first()
+        .ok_or(errors::ApiErrorResponse::IncorrectPaymentMethodConfiguration)
+        .attach_printable(
+            "No eligible connector found for the network transaction id based mit flow",
+        )?;
+
+    // Set `NetworkMandateId` as the MandateId
+    payment_data.set_mandate_id(payments_api::MandateIds {
+        mandate_id: None,
+        mandate_reference_id: Some(mandate_reference_id),
+    });
+
+    // Set the card details received in the recurring details within the payment method data.
+    payment_data.set_payment_method_data(Some(
+        hyperswitch_domain_models::payment_method_data::PaymentMethodData::CardDetailsForNetworkTransactionId(card_details),
+    ));
+
+    // Construct routing data using the selected connector information, which will be set in the payment attempt.
+    let routing_data = storage::RoutingData {
```

Comment:
Is setting this data in attempt is really needed? because we are storing the necessary information in attempt directly and why another source of truth is required?